lib/ash/type/type.ex

```diff
@@ -812,31 +812,35 @@ defmodule Ash.Type do
         term
         |> Enum.with_index()
         |> Enum.reduce({[], []}, fn {item, index}, {items, errors} ->
-          if is_nil(item) && not nil_items? do
-            {[item | items], [[message: "no nil values", index: index] | errors]}
-          else
-            if type.custom_apply_constraints_array?() do
-              {[item | items], errors}
+          if type.custom_apply_constraints_array?() do
+            if is_nil(item) && not nil_items? do
```

Comment:
Is the check duplication OK, or is there a better/cleaner way to do this? Still kinda new to Elixir so I don't master the best practices yet :sweat_smile:

Reply:
Well I'm working on the next PR (`remove_nil_items?`) and created a private function in there to centralize the duplicated code :+1:

Reply:
A little duplication never hurt anybody 😆